tests/test_metrics.py

```diff
@@ -17,6 +24,91 @@
 
 logger = get_logger()
 
+# values of inputs that are common to grouped_mean type InstanceMetric
+GROUPED_INSTANCE_PREDICTIONS = [
+    "A B",
+    "BC D",
+    "C",
+    "123",
+    "BCD",
+    10,
+    "  BD",
+    "AB",
+    "I am a dog",
+    "AB C",
+    "AB 1",
+    "GMA",
+    0.123,
+    "BD",
+    "abc",
+]
+
+GROUPED_INSTANCE_REFERENCES = [
+    ["B", "AB", "A"],
+    ["A", "BC D", "BC DF"],
+    ["c", " C"],
+    [13, 23, 234],
+    ["  ", " BD", " BDA"],
+    [1, 10, 100],
+    ["A", "B", "BD"],
+    ["ABC", "ab", "BC"],
+    ["I am a person", "I AM A DOG", "ABC"],
+    ["AB CD", "AB", "ab"],
+    ["AB 1", "AB1"],
+    [" GMA 123", "GMA"],
+    ["123", 0.12],
+    ["BDE", "BCE", "bdefs"],
+    [" abcdefg", "AB", "abcd"],
+]
+
+GROUPED_INSTANCE_PREDICTIONS_SHORT = [
```

Comment:
a question: why are there two sets of predictions and references (regular an `short`)?

Reply:
This was something I did when I tested other precision/recall based scores like F1 which did some kind of averaging over the observed labels, which is not "interesting" if there are not only a few unique labels.  But I just tested it with the original sets, which still yield meaningful values.  So I will change this and the tests to use the same set of predictions.  If we had a F1-type InstanceMetric (as opposed to GlobalMetric), this would be relevant.